src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -411,6 +412,8 @@ where
         &mut self,
         commands: Vec<(GlobalId, Vec<Update<Self::Timestamp>>, Self::Timestamp)>,
     ) -> Result<(), StorageError> {
+        let futs = FuturesUnordered::new();
```

Comment:
> join_all will switch to the more powerful [FuturesOrdered](https://docs.rs/futures/0.3.21/futures/stream/struct.FuturesOrdered.html) for performance reasons if the number of futures is large. You may want to look into using it or it’s counterpart [FuturesUnordered](https://docs.rs/futures/0.3.21/futures/stream/struct.FuturesUnordered.html) directly.

I was about to leave a comment about just using `join_all` instead, which seems to be much simpler when I found this bit in the rustdoc for it. is that why you're using FuturesUnordered here over join_all? if so, it deserves a comment